### PR TITLE
Workflow

### PR DESCRIPTION
**Pull request by yozhgoor**
I have add Github actions for PRs, main branch and release.

### CODE REVIEW

**Comment by yozhgoor** [comment]
Say me if you want to add the Windows tests

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: main
+
+on:
+  push:
+    branches: [ main ]
+  schedule:
+    - cron: 0 0 1 * *
```
Comment (o0Ignition0o) [inline .github/workflows/main.yml:7]: oh thats a nice trick, hadn't thought of that! 🎉

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      uses: actions-rs/cargo@v1
+      with:
+        command: fmt
+        args: --all -- --check
+
+    - name: clippy
+      uses: actions-rs/clippy-check@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        args: -- -D warnings
```
Comment (o0Ignition0o) [inline .github/workflows/pr.yml:35]: has this been discussed yet ? poke @cbonaudo @scrabsha

**Review by o0Ignition0o — APPROVED** [review]

File: .github/workflows/pr.yml (hunk 2 above)
Comment (cbonaudo) [reply .github/workflows/pr.yml:35]: seems cool, but we might fix (if any) the lints beforehand) or if we don't mind having failing CI for a while

Comment (scrabsha) [inline .github/workflows/pr.yml:35]: Please add `--tests` here, so that we get lints for tests too

**Review by scrabsha — CHANGES_REQUESTED** [review]

File: .github/workflows/pr.yml (hunk 2 above)
Comment (yozhgoor) [reply .github/workflows/pr.yml:35]: What do you think of 
> in order to also check tests and non-default crate features, use 
> `cargo clippy --all-targets --all-features -- -D warnings`

Comment (scrabsha) [reply .github/workflows/pr.yml:35]: That's perfect :)

**Review by scrabsha — APPROVED** [review]
lgtm, thanks!

**Pull request merged**